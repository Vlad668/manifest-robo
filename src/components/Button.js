
import {useState} from 'react'

const Button = () => {
    const [count, setCount] = useState(0);
    return (
        <div key='12'>
        <p key='8'>{count}</p>
        <button id='counter' onClick={() => setCount(count + 1)}>Click me</button>
        </div>
    )
}

export default Button