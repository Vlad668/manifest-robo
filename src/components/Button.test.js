import React from 'react';
import "@testing-library/jest-dom/extend-expect";
import { create } from "react-test-renderer"
import {shallow} from 'enzyme'
import Button from './Button'






  it('should set the count value on click', () => {
    
    const wrapper = shallow(<Button/>)
    
    wrapper.find('[id="counter"]').simulate('click');
    wrapper.find('[id="counter"]').simulate('click');
    wrapper.find('[id="counter"]').simulate('click');

    expect((wrapper.find('p').text())).toEqual('3')
    expect(create(wrapper).toJSON()).toMatchSnapshot();
    // expect(create(wrapper.find('[id="counter"]')).toJSON()).toEqual('counter')
    
  });

  
//   it('Counter: renders correctly', () => {
//      const app = create(<Button />);
//       expect(app.toJSON()).toMatchSnapshot()
//     });



  