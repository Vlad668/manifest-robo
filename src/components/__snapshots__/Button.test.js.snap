// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should set the count value on click 1`] = `
<div>
  <p>
    3
  </p>
  <button
    id="counter"
    onClick={[Function]}
  >
    Click me
  </button>
</div>
`;
